README.md

```diff
@@ -9,21 +9,621 @@ JavaScript Inheritance Patterns
+not foolproof.
+
+However, the even bigger problem with constructor functions and `new`
+shows up when we try to inherit from another object and take advantage
+of the delegation features in the prototype system.
+
+When an object is created, its prototype is set to the
+object it was created from. In the case of a new object created with the
+`{}` literal, that prototype will be `Object.prototype`.
+
+But what do we know about functions in JavaScript? They are also
+objects! So when we create an object from a constructor function, we are
+really making that function the object's `prototype`!
+
+![kramer mind blown](http://i.giphy.com/OK27wINdQS5YQ.gif)
+
+Let's look at our example one more time.
+
+```js
+// Rectangle constructor
+function Rectangle(sides, width, height) {
+  this.sides = sides;
+  this.width = width;
+  this.height = height;
+  this.area = function() {
+    return this.width * this.height;
+  }
+  this.perimeter = function() {
+    return (this.width + this.height) * 2;
+  }
+}
+
+// Square constructor
+function Square(sides, length) {
+  Rectangle.call(this, sides, length, length);
+}
+
+// rect prototype becomes Rectangle()
+var rect = new Rectangle(4, 3, 5);
+// square prototype becomes Square()
+var square = new Square(4, 2);
+
+// this works because the rect prototype is still Rectangle
+Rectangle.prototype.internalAngles = 90;
+rect.internalAngles;
+
+// but the square prototype is not Rectangle
+square.internalAngles;
+```
+
+Even though we call the `Rectangle` constructor function inside the
+`Square` constructor function, the function that actually *creates*
+`square` is `new Square()`, so the prototype for the square is
+`Square.prototype`, meaning that we haven't inherited from
+`Rectangle` at all, we've just borrowed its constructor function to make our own
+object, and made copies of its properties and methods, but haven't taken
+advantage of prototyping.
+
+### Object.create()
+
+Fortunately, we have `Object.create()` to rescue us from these problems
+with `new`. Rather than try to force class-style inheritance with `new`,
+we'll use the pure object-to-object prototype system that JavaScript is
+built on.
+
+In order to create a `Square` that truly inherits from `Rectangle`, we'll need
+to make a few changes to how we put our objects together.
+
+We need to convert our constructor function to a simple object
+definition, creating the "prototype" for all rectangles to come:
+
+```js
+var rectangle = {
+  sides: 4,
+  width: 5,
+  height: 7,
+  area: function() {
+  	return this.width * this.height;
+  },
+  perimeter: function () {
+  	return (this.width + this.height) * 2;
+  }
+}
+```
+
+We have created a perfectly functional `rectangle` object, and not only
+can we use it, but we can use it to create other rectangle-like objects.
+Let's try a few things:
+
+```js
+var rect = Object.create(rectangle);
+
+console.log(rect);
+// empty Object {}
+
+console.log(rect.sides);
+// 4
+
+console.log(rect.__proto__);
```

Comment:
Should we use `getPrototypeOf()` instead? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/proto